tests/external_executor/external_executor.py

```diff
@@ -481,8 +481,25 @@ def test_logging_executor(network, args):
 
             r = c.post("/app/log/public", {"id": log_id, "msg": log_msg})
             assert r.status_code == 200
-
             r = c.get(f"/app/log/public?id={log_id}")
+
+            assert r.status_code == 200
+            assert r.body.json()["msg"] == log_msg
+
+            # post to private table
+            r = c.post("/app/log/private", {"id": log_id, "msg": log_msg})
+            assert r.status_code == 200
+            tx_id = r.headers.get("x-ms-ccf-transaction-id")
+
+            # make a historical query
+            time.sleep(3)
+            headers = {"x-ms-ccf-transaction-id": tx_id}
+            r = c.get(f"/app/log/private/historical?id={log_id}", headers=headers)
+            assert r.status_code == 202
+
+            time.sleep(3)
```

Comment:
In general, we shouldn't have any sleeps in our tests outside of a retry loop on a well defined condition. For asynchronous operations, sleeping is either brittle (when too short), or wasteful (when too long). A retry loop with a timeout is efficient (passes as soon as the condition is met) and robust (the timeout can be set to a value so large that it isn't triggered by temporarily slow I/O or network).

Reply:
Makes sense, I've now modified it to a retry loop.